staging/src/k8s.io/apiserver/pkg/util/shufflesharding/shufflesharding_test.go

```diff
@@ -272,33 +275,73 @@ func TestShuffleAndDealToSliceWithValidation(t *testing.T) {
 
 func BenchmarkShuffleAndDealToSlice(b *testing.B) {
 	hashValueBase := math.MaxUint64 / uint64(b.N)
-	deckSize, handSize := int32(512), int32(8)
+	deckSize, handSize := 512, 8
 	for i := 0; i < b.N; i++ {
 		_ = ShuffleAndDealToSlice(hashValueBase*uint64(i), deckSize, handSize)
 	}
 }
 
+// ff computes the falling factorial `n!/(n-m)!` and requires n to be
+// positive and m to be in the range [0, n] and requires the answer to
+// fit in an int
+func ff(n, m int) int {
+	ans := 1
+	for f := n; f > n-m; f-- {
+		ans *= f
+	}
+	return ans
+}
+
 func TestUniformDistribution(t *testing.T) {
-	deckSize, handSize := int32(128), int32(3)
-	handCoordinateMap := make(map[int]int)
+	const spare = 64 - maxHashBits
+	tests := []struct {
+		deckSize, handSize int
+		hashMax            int
+	}{
+		{64, 3, 1 << uint(math.Ceil(math.Log2(float64(ff(64, 3))))+spare)},
+		{128, 3, ff(128, 3)},
+		{128, 3, 3 * ff(128, 3)},
```

Comment:
If {128, 3, ff(128,3)} pass, {128, 3, 3 *ff(128, 3)} will pass either, so I think this case is duplication.

Reply:
My thinking for the `{128, 3, 3 * ff(128, 3)}` test case is that by exploring a larger hash value space this gives the implementation more opportunities to mess up.